### PR TITLE
Property documentation: Rework configuration reference

### PR DESCRIPTION
**Pull request by cweiske**
Resolves: https://github.com/EugenMayer/docker-sync/issues/39

### CODE REVIEW

**Comment by EugenMayer** [comment]
wow that is quiet some chunk of work, thank you so much. right now trying to understand how to see this pr on readthedocs to see how  this would view there..

**Comment by cweiske** [comment]
I used a local sphinx installation to render it

**Comment by EugenMayer** [comment]
could you just do an additional commit onto this pr, i just activated pr builds with readthedocs but it needs to be retriggered

File: docs/getting-started/configuration.rst [hunk 1]
```diff
@@ -21,124 +21,415 @@ Below are all the available options, simple examples can be found in the docker-
...
+        sync_excludes: ['ignored_folder', '.ignored_dot_folder']
 
-      # optional, default: docker-compose-dev.yml if you like, you can set a custom location (path) of your compose file. Do not set it, if you do not want to use it at all
 
-      # if its there, it gets used, if you name it explicitly, it HAS to exist
-      # HINT: you can also use this as an array to define several compose files to include. Order is important!
-      compose-dev-file-path: 'docker-compose-dev.yml'
+General options
+---------------
+Configuration options below the top-level ``options`` key.
```
Comment (EugenMayer) [inline docs/getting-started/configuration.rst:41]: we should add a visual example here, what do you think?

File: docs/getting-started/configuration.rst [hunk 2]
```diff
@@ -21,124 +21,415 @@ Below are all the available options, simple examples can be found in the docker-
...
 
-      # optional, default: pwd, root directory to be used when transforming sync src into absolute path, accepted values: pwd (current working directory), config_path (the directory where docker-sync.yml is found)
-      project_root: 'pwd'
+default:
+  ``auto``
+
+possible values:
+  - ``docker-sync``
+  - ``thor``
+  - ``auto``
```
Comment (EugenMayer) [inline docs/getting-started/configuration.rst:58]: maybe we should use a different format here, e.g. a table with 2 columns,first col for the possible values, the second is the default? what is your opinion?

Comment (cweiske) [reply docs/getting-started/configuration.rst:58]: I use the "possible values" to explain the option sometimes, so having them separate is helpful IMO

File: docs/getting-started/configuration.rst (hunk 1 above)
Comment (cweiske) [reply docs/getting-started/configuration.rst:41]: Done.

File: docs/getting-started/configuration.rst (hunk 2 above)
Comment (EugenMayer) [reply docs/getting-started/configuration.rst:64]: sure, its more about table vs list

**Comment by EugenMayer** [comment]
Seems like the readthedocs PR build works either for only local branches or for PRs newly created. If you like, just recreate your PR - mine seems to work https://github.com/EugenMayer/docker-sync/pull/745

I added a docker based variant to work on the docs in `docs/README.md`

**Comment by cweiske** [comment]
Continuing in #746.

**Pull request closed without merging**